src/api/worker/facades/lazy/CalendarFacade.ts

```diff
@@ -384,26 +379,29 @@ export class CalendarFacade {
 	 * Queries the event using the uid index. The index is stored per calendar so we have to go through all calendars to find matching event.
 	 * We currently only need this for calendar event updates and for that we don't want to look into shared calendars.
 	 */
-	getEventByUid(uid: string): Promise<CalendarEvent | null> {
+	async getEventByUid(uid: string): Promise<CalendarEvent | null> {
 		const calendarMemberships = this.userFacade.getLoggedInUser().memberships.filter((m) => m.groupType === GroupType.Calendar && m.capability == null)
-
-		return asyncFindAndMap(calendarMemberships, (membership) => {
-			return this.entityClient
-				.load(CalendarGroupRootTypeRef, membership.group)
-				.then(
-					(groupRoot) =>
-						groupRoot.index &&
-						this.entityClient.load<CalendarEventUidIndex>(CalendarEventUidIndexTypeRef, [groupRoot.index.list, uint8arrayToCustomId(hashUid(uid))]),
-				)
-				.catch(ofClass(NotFoundError, () => null))
-				.catch(ofClass(NotAuthorizedError, () => null))
-		}).then((indexEntry) => {
-			if (indexEntry) {
-				return this.entityClient.load<CalendarEvent>(CalendarEventTypeRef, indexEntry.calendarEvent).catch(ofClass(NotFoundError, () => null))
-			} else {
-				return null
+		for (const membership of calendarMemberships) {
```

Comment:
this will conflict with rescheduling, we also rewrote this function

Reply:
I guess we can just integrate the fix into the rescheduling PR then, it's not a lot.